### PR TITLE
[WIP] refactor scheduling policy interface

### PR DESCRIPTION
**Pull request by wumuzi520**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/raylet/scheduling/policy/scheduling_policy.h [hunk 1]
```diff
@@ -94,9 +97,46 @@ class ISchedulingPolicy {
...
+      SchedulingContext *context = nullptr) override {
+    size_t success_count = 0;
+    std::vector<scheduling::NodeID> selected_nodes;
+    selected_nodes.reserve(resource_request_list.size());
+    for (auto &resource_request : resource_request_list) {
+      auto node_id = Schedule(*resource_request, options, context);
+      if (!node_id.IsNil()) {
+        ++success_count;
+      }
+      selected_nodes.emplace_back(node_id);
```
Comment (scv119) [inline src/ray/raylet/scheduling/policy/scheduling_policy.h:126]: we can add a todo here comments that we should deduct the resource temporarily

File: src/ray/raylet/scheduling/policy/composite_scheduling_policy.cc [hunk 2]
```diff
@@ -24,27 +24,28 @@ namespace ray {
...
-  default:
-    RAY_LOG(FATAL) << "Unsupported scheduling type: "
-                   << static_cast<typename std::underlying_type<SchedulingType>::type>(
-                          options.scheduling_type);
+    const ResourceRequest &resource_request,
+    SchedulingOptions options,
+    SchedulingContext *context /*= nullptr*/) {
+  auto result = Schedule({&resource_request}, options, context);
+  if (result.status.IsSuccess()) {
+    RAY_CHECK(result.selected_nodes.size() == 1);
```
Comment (raulchen) [inline src/ray/raylet/scheduling/policy/composite_scheduling_policy.cc:32]: This check can be moved to inside the "Schedule" function. if success, check result size == input size

File: src/ray/raylet/scheduling/policy/scheduling_policy.h [hunk 3]
```diff
@@ -37,6 +40,8 @@ struct SchedulingResultStatus {
     INFEASIBLE = 1,
     // Scheduling successful.
     SUCCESS = 2,
+    // Only part of nodes succeed when batch scheduling.
```
Comment (raulchen) [inline src/ray/raylet/scheduling/policy/scheduling_policy.h:43]:
```suggestion
    // Only part of the requested resources succeed when batch scheduling.
```

File: src/ray/raylet/scheduling/policy/composite_scheduling_policy.cc (hunk 2 above)
Comment (wumuzi520) [reply src/ray/raylet/scheduling/policy/composite_scheduling_policy.cc:32]: > This check can be moved to inside the "Schedule" function. if success, check result size == input size

I don't think so, this is a single schedule method, it is more clear to put `check` in this method directly.

**Pull request closed without merging**